src/librawspeed/decoders/IiqDecoder.cpp

```diff
@@ -242,6 +260,153 @@ void IiqDecoder::DecodePhaseOneC(const std::vector<IiqStrip>& strips,
     DecodeStrip(strip, width, height);
 }
 
+void IiqDecoder::CorrectPhaseOneC(ByteStream meta_data, uint32 split_row,
+                                  uint32 split_col) {
+  meta_data.skipBytes(8);
+  const uint32 bytes_to_entries = meta_data.getU32();
+  meta_data.setPosition(bytes_to_entries);
+  const uint32 entries_count = meta_data.getU32();
+  meta_data.skipBytes(4);
+
+  // this is how much is to be read for all the entries
+  ByteStream entries(meta_data.getStream(entries_count, 12));
+  meta_data.setPosition(0);
+
+  bool quadrant_multiplier_applied = false;
+
+  for (uint32 entry = 0; entry < entries_count; entry++) {
+    const uint32 tag = entries.getU32();
+    const uint32 len = entries.getU32();
+    const uint32 offset = entries.getU32();
+
+    switch (tag) {
+    case 0x431:
+      if (!quadrant_multiplier_applied) {
```

Comment:
Are there cases (raws) where there more than one `0x431` tag?

Reply:
Oddly enough, there seem to be multiple of them in all my RAWs.  I'm not sure how the manufacturer's software deals with this, but dcraw just applies whatever is the first quadrant correction tag (of any of the three types it handles) in the file

Reply:
> dcraw just applies whatever is the first quadrant correction tag

So just return after the `CorrectQuadrantMultipliersCombined()` call?

Reply:
I could do that for now if you want, it just won't make sense once there's more corrections in here that can apply to the same image

Reply:
> once there's more corrections in here that can apply to the same image

I have already replied to that in https://github.com/darktable-org/rawspeed/pull/117#issuecomment-347901618
The decision of *what* is rawspeed's job is case-by-case based.

Reply:
Will you still want the bad pixel corrections in RS?  Those appear alongside the quadrant multiplier

Reply:
Yes, bad pixel are fair game :)
But that is for future pr's.